src/Tgstation.Server.Host/Components/Chat/Providers/IrcProvider.cs

```diff
@@ -487,6 +457,37 @@ await Task.Factory.StartNew(
 			}
 		}
 
+		/// <summary>
+		/// Register the client on the network.
+		/// </summary>
+		/// <param name="client">IRC client.</param>
+		/// <param name="nickname">Nickname.</param>
+		/// <returns><see cref="Task"/> that resolves when registration has been completed. </returns>
+		/// <exception cref="TimeoutException">If the IRC server fails to respond.</exception>
+		private async Task Login(IrcFeatures client, string nickname)
+		{
+			var promise = new TaskCompletionSource<object>();
+
+			void Callback(object? sender, EventArgs e)
+			{
+				Logger.LogTrace("IRC Registered.");
+				promise.TrySetResult(e);
+			}
+
+			client.OnRegistered += Callback;
+
+			client.Login(nickname, nickname, 0, nickname);
+
+			var completed = await Task.WhenAny(promise.Task, Task.Delay(30 * 1000));
+			if (completed == promise.Task)
+			{
+				client.OnRegistered -= Callback;
+				return;
+			}
+
+			throw new TimeoutException("Timed out waiting for IRC registration.");
```

Comment:
```suggestion
			using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
			cts.CancelAfter(TimeSpan.FromSeconds(30));
			try
			{
				await promise.Task.WaitAsync(cts.Token);
			}
			catch (OperationCanceledException ex)
			{
				throw new JobException("Timed out waiting for IRC registration.", ex);
			}
```

Pass `CancellationToken` through from `Connect`.